### PR TITLE
Rename acme_diags to e3sm_diags

### LINKED ISSUE #458: rename `acme_diags` to `e3sm_diags`
I think it would make sense to rename the `acme_diags` package to `e3sm_diags`.  You could also continue to make an `acme_diags` shadow package that points to `e3sm_diags` but is deprecated and will be removed in a future release.  Here is an example from the `pyevtk` project, which has a deprecated `evtk` package:
https://github.com/pyscience-projects/pyevtk/blob/master/evtk/__init__.py

### PR DESCRIPTION
**Pull request by tomvothecoder**
This PR renames the source directory `acme_diags` to `e3sm_diags` using PyCharm's refactoring capabilities to automatically update references.
    - Rename corresponding unit testing directory
    - Update references in docs

- Closes #458

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -114,14 +114,14 @@ def get_all_files_in_dir(directory, pattern):
...
+    scripts=["e3sm_diags/acme_diags_driver.py"],
+    packages=find_packages(include=["e3sm_diags", "e3sm_diags.*"]),
     data_files=data_files,
     entry_points={
         "console_scripts": [
-            "e3sm_diags=acme_diags.acme_diags_driver:main",
-            "acme_diags=acme_diags.acme_diags_driver:main",
-            "e3sm_diags_vars=acme_diags.acme_diags_vars:main",
+            "e3sm_diags=e3sm_diags.acme_diags_driver:main",
+            "e3sm_diags=e3sm_diags.acme_diags_driver:main",
```
Comment (xylar) [inline setup.py:123]:
```suggestion
            "e3sm_diags=e3sm_diags.acme_diags_driver:main",
```

Comment (xylar) [reply setup.py:123]: `acme_diags_driver` should also be renamed to `e3sm_diags_driver`.

Comment (xylar) [reply setup.py:123]: And `acme_diags_vars` should become `e3sm_diags_vars`.

**Comment by tomvothecoder** [comment]
@xylar @chengzhuzhang This PR is ready for review. The diff is pretty large, but most of it consists of file renames without changes due to the directory name change.

File: docs/source/defining-parameters.rst [hunk 2]
```diff
@@ -222,8 +222,8 @@ We can do the following:
 
 .. code:: python
 
-    reference_data_path = '/global/cfs/cdirs/e3sm/acme_diags/obs_for_acme_diags/'
-    test_data_path = '/global/cfs/cdirs/e3sm/acme_diags/test_model_data_for_acme_diags/'
+    reference_data_path = '/global/cfs/cdirs/e3sm/e3sm_diags/obs_for_acme_diags/'
```
Comment (chengzhuzhang) [inline docs/source/defining-parameters.rst:225]: The data paths in docs/examples should remain the same.

File: docs/source/defining-parameters.rst [hunk 3]
```diff
@@ -222,8 +222,8 @@ We can do the following:
...
+    test_data_path = '/global/cfs/cdirs/e3sm/e3sm_diags/test_model_data_for_acme_diags/'
```
Comment (chengzhuzhang) [inline docs/source/defining-parameters.rst:226]: same as above

**Comment by chengzhuzhang** [comment]
Hey Tom, Thanks for the PR. I notice that the string `acme_diags` in data paths (where store obs and test data, e.x. reference_data_path = '/global/cfs/cdirs/e3sm/e3sm_diags/obs_for_acme_diags/')
@chengzhuzhang ) are also changed. We should either change back these strings, or generate symlinks with new names for these directories on different servers. The latter option may cause some confusion in directory, so I slightly lean towards keeping the `acme_diags` names in data path names.

File: docs/source/defining-parameters.rst (hunk 2 above)
Comment (xylar) [reply docs/source/defining-parameters.rst:225]: @chengzhuzhang, wouldn't the better solution be to rename these paths as well, and put a symlink to the old `acme_diags` version?  It seems confusing to keep the old name anywhere unless it's absolutely necessary.

**Comment by xylar** [comment]
@chengzhuzhang, I see you had the same thought I did about symlinks.  I do think renaming to `e3sm_diags` on the data server and having a symlink to `acme_diags` (for backwards compatibility) is the less confusing solution.  But since it's not my code I'll leave it up to you.

**Comment by xylar** [comment]
@tomvothecoder and @chengzhuzhang, I tried testing this on Chrysalis by following the documentation.  I ran into a few difficulties with the docs that I posted issues for.

I'm not particularly expert at running `e3sm_diags` so I followed https://e3sm-project.github.io/e3sm_diags/_build/html/master/dev_guide/testing.html#complete-run-test.  But I got errors there, too:

```
E
======================================================================
ERROR: test_complete_run (tests.complete_run.TestCompleteRun)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/gpfs/fs1/home/ac.xylar/e3sm_work/e3sm_diags/458-rename-acme_diags/tests/complete_run.py", line 22, in test_complete_run
    with open(expected_images_file) as f:
FileNotFoundError: [Errno 2] No such file or directory: '/lcrc/group/e3sm/public_html/e3sm_diags_test_data/unit_test_complete_run/expected/image_list_all_sets.txt'

----------------------------------------------------------------------
Ran 1 test in 472.384s

FAILED (errors=1)
```

These don't seem directly related to this PR.  I can make another issue about these if you like.

I could use a suggestion for what testing would be more helpful.

**Comment by chengzhuzhang** [comment]
> @chengzhuzhang, I see you had the same thought I did about symlinks. I do think renaming to `e3sm_diags` on the data server and having a symlink to `acme_diags` (for backwards compatibility) is the less confusing solution. But since it's not my code I'll leave it up to you.

Thank you, Xylar. Renaming the path and having a symlink to `acme_diags`for backwards compatibility is smart. I think let's go with this.

File: docs/source/defining-parameters.rst (hunk 2 above)
Comment (tomvothecoder) [reply docs/source/defining-parameters.rst:225]: wontfix based on decision to use symlinks from `acme_diags` to `e3sm_diags`.

File: docs/source/defining-parameters.rst (hunk 3 above)
Comment (tomvothecoder) [reply docs/source/defining-parameters.rst:226]: wontfix based on decision to use symlinks from `acme_diags` to `e3sm_diags`.

**Comment by tomvothecoder** [comment]
> Thank you, Xylar. Renaming the path and having a symlink to acme_diagsfor backwards compatibility is smart. I think let's go with this.

Sounds like this PR is good to go. @chengzhuzhang I'll open up a new issue to update symlinks and assign it to you since you're more familiar with the directory structure.

> These don't seem directly related to this PR. I can make another issue about these if you like.
> 
> I could use a suggestion for what testing would be more helpful.

@xylar yeah, that error isn't related to this PR since you're running the complete test suite. 
I ran the complete test suite and received the same error. I also confirmed that the file in your error message does not exist: `'/lcrc/group/e3sm/public_html/e3sm_diags_test_data/unit_test_complete_run/expected/image_list_all_sets.txt'`

If could open up a new issue documenting the error, that would be great. @forsyth2 usually handles the complete test runs.

The CI/CD workflow runs a subset of the integration/system tests ([here](https://e3sm-project.github.io/e3sm_diags/_build/html/master/dev_guide/testing.html#unit-and-integration-tests)), which is why this PR is still passing.

**Comment by forsyth2** [comment]
> I ran the complete test suite and received the same error. I also confirmed that the file in your error message does not exist: '/lcrc/group/e3sm/public_html/e3sm_diags_test_data/unit_test_complete_run/expected/image_list_all_sets.txt'

Somehow, the image list got deleted. I just ran `find . -type f -name '*.png' > ../image_list_all_sets.txt` from `/lcrc/group/e3sm/public_html/e3sm_diags_test_data/unit_test_complete_run/expected/all_sets`, so that gets rid of that error.

However, I'm now debugging a `FileNotFoundError: [Errno 2] No such file or directory: './v2_4_0_all_sets/all_sets_v250_20210608_1878984/cosp_histogram/MODIS-COSP/MODISCOSP-COSP_HISTOGRAM_MODIS-JJA-global.png'` error....

**Comment by forsyth2** [comment]
> However, I'm now debugging a FileNotFoundError: [Errno 2] No such file or directory: './v2_4_0_all_sets/all_sets_v250_20210608_1878984/cosp_histogram/MODIS-COSP/MODISCOSP-COSP_HISTOGRAM_MODIS-JJA-global.png' error....

@chengzhuzhang The above error seems to be because `all_sets_v250_20210608_1878984` (an exact copy of `/lcrc/group/e3sm/public_html/e3sm_diags_test_data/unit_test_complete_run/expected/previous_output/all_sets_v250_20210608_1878984`) is in `/lcrc/group/e3sm/public_html/e3sm_diags_test_data/unit_test_complete_run/expected/all_sets`. I can't `rm -rf` it though because the permissions are `drwxr-xr-x 17 ac.zhang40  E3SM`

I think we need to add a step to change permissions on the directions to replace the expected images (at https://e3sm-project.github.io/e3sm_diags/_build/html/master/dev_guide/testing.html#complete-run-test). Something like `chmod -R g+wX /lcrc/group/e3sm/public_html/e3sm_diags_test_data/unit_test_complete_run/expected/all_sets`

**Comment by xylar** [comment]
@forsyth2, this seems like something to create an issue for and discuss there.

**Comment by forsyth2** [comment]
> @forsyth2, this seems like something to create an issue for and discuss there.

I created #506

**Pull request merged**